### PR TITLE
refactor: Bootstrap to AntD - Label

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
- Migrates Label component from Bootstrap to AntD.
- Removes unused properties.
- Renames `bsStyle` property to `type` and add typescript checking.
- Improves CSS readability.
- Modifies storybook to use controls instead of handles.
- Updates tests.

See: #10254

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-01-26 at 2 09 26 PM
Screen Shot 2021-01-26 at 2 28 45 PM
Screen Shot 2021-01-26 at 2 10 08 PM
Screen Shot 2021-01-26 at 2 27 07 PM
Screen Shot 2021-01-26 at 2 27 29 PM
Screen Shot 2021-01-26 at 2 12 56 PM
Screen Shot 2021-01-26 at 2 10 59 PM
Screen Shot 2021-01-26 at 2 26 49 PM

@rusackas @junlincc

### TEST PLAN
1 - Open any screen that contains a label (explore, profile, dashboards, etc.)
2 - All labels should have the same theme and behavior
3 - Additional testing can be done using Storybook

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Label/Label.test.tsx [hunk 1]
```diff
@@ -34,23 +34,13 @@ describe('Label', () => {
   it('works with an onClick handler', () => {
     const mockAction = jest.fn();
     wrapper = mount(<Label onClick={mockAction} />);
-    wrapper.find('.label').simulate('click');
+    wrapper.find(Label).simulate('click');
     expect(mockAction).toHaveBeenCalled();
   });
 
   // test stories from the storybook!
-  it('renders all the sorybook gallery variants', () => {
```
Comment (rusackas) [inline superset-frontend/src/components/Label/Label.test.tsx:42]: oops!

File: superset-frontend/src/components/Label/index.tsx [hunk 2]
```diff
@@ -17,149 +17,98 @@
...
-import { Label as BootstrapLabel } from 'react-bootstrap';
-import { styled } from '@superset-ui/core';
-import cx from 'classnames';
+import { Tag } from 'src/common/components';
+import { useTheme } from '@superset-ui/core';
 
-export type OnClickHandler = React.MouseEventHandler<BootstrapLabel>;
+export type OnClickHandler = React.MouseEventHandler<HTMLElement>;
+
+export type BsStyle =
```
Comment (rusackas) [inline superset-frontend/src/components/Label/index.tsx:25]: The name BsStyle/bsStyle is a reference to bootstrap. Now that it's out of the picture, we can rename this globaly to something more general, like type/variant/style/etc.

File: superset-frontend/src/components/Label/index.tsx [hunk 3]
```diff
@@ -17,149 +17,98 @@
...
+    } else if (bsStyle === 'warning') {
+      baseColor = warning;
+    } else if (bsStyle === 'danger') {
+      baseColor = error;
+    } else if (bsStyle === 'info') {
+      baseColor = info;
+    } else if (bsStyle === 'secondary') {
+      baseColor = secondary;
+    } else {
+      baseColor = primary;
```
Comment (rusackas) [inline superset-frontend/src/components/Label/index.tsx:79]: This is all so much cleaner. Love it!

**Review by rusackas — CHANGES_REQUESTED** [review]
This is looking fantastic! Ready to approve, but would like to change that bsStyle prop name.

**Comment by adam-stasiak** [comment]
I can see that Database label is a little bit moved. On master is straight.
Zrzut ekranu 2021-01-27 o 17 12 48

**Comment by michael-s-molina** [comment]
> I can see that Database label is a little bit moved. On master is straight.
> Zrzut ekranu 2021-01-27 o 17 12 48
> No other info found. @junlincc Could you set priority on this?

@adam-stasiak Thank you so much for reviewing! I'll wait until the end of today to see if you find anything else so I can commit the fixes only once.

**Comment by michael-s-molina** [comment]
@adam-stasiak Thank you so much for reviewing! Fixed!

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**